src/cert/mod.rs

```diff
@@ -109,6 +114,168 @@ pub async fn upload_new_cert_ref(
     Ok(cert_ref)
 }
 
+fn format_cert_for_multi_select(cert: &CageSigningCert) -> String {
+    let name = cert.name().unwrap_or_else(|| "".to_string());
+    let cert_hash = cert.cert_hash();
+    let not_after = cert
+        .not_after()
+        .and_then(|time| format_expiry_time(&time).ok())
+        .unwrap_or_else(|| "".to_string());
+
+    format!("{} {} ({})", name, cert_hash, not_after)
+}
+
+fn format_expiry_time(expiry_time: &str) -> Result<String, CertError> {
+    let dt = DateTime::parse_from_rfc3339(expiry_time).map_err(CertError::TimstampParseError)?;
+
+    let now = Local::now();
+    let duration = dt.signed_duration_since(now);
+
+    let formatted_duration = if duration.num_hours() < 0 {
+        "Expired".to_string()
+    } else if duration.num_hours() >= 24 {
+        format!("Expires in {} days", duration.num_days())
+    } else {
+        format!("Expires in {} hours", duration.num_hours())
+    };
+
+    Ok(formatted_duration)
+}
+
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
+struct CertWithFormattedString {
+    cert: CageSigningCert,
+    formatted: String,
+    locked: bool,
+}
+
+impl CertWithFormattedString {
+    fn new(cert: &CageSigningCert, locked: bool) -> Self {
+        Self {
+            formatted: format_cert_for_multi_select(cert),
+            cert: cert.clone(),
+            locked,
+        }
+    }
+}
+
+async fn get_certs_for_selection(
+    cage_api: api::cage::CagesClient,
+    cage_uuid: &str,
+) -> Result<Vec<CertWithFormattedString>, CertError> {
+    let available_certs = match cage_api.get_signing_certs().await {
+        Ok(res) => res.certs,
+        Err(e) => {
+            log::error!("Error getting cage signing cert refs — {:?}", e);
+            return Err(CertError::ApiError(e));
+        }
+    };
+
+    let locked_certs = match cage_api.get_cage_locked_signing_certs(cage_uuid).await {
+        Ok(certs) => certs,
+        Err(e) => {
+            log::error!("Error getting cage signing cert — {:?}", e);
+            return Err(CertError::ApiError(e));
+        }
+    };
+
+    let locked_cert_uuids = locked_certs
+        .iter()
+        .map(|cert| cert.uuid())
+        .collect::<Vec<&str>>();
+
+    let available_stripped: Vec<&CageSigningCert> = available_certs
+        .iter()
+        .filter(|cert| !locked_cert_uuids.contains(&cert.uuid()))
+        .collect();
+
+    let available_formatted = available_stripped
+        .iter()
+        .map(|cert| CertWithFormattedString::new(cert, false))
+        .collect::<Vec<CertWithFormattedString>>();
```

Comment:
Worth doing this in one step?
```suggestion
    let available_stripped: Vec<&CageSigningCert> = available_certs
        .iter()
        .filter(|cert| !locked_cert_uuids.contains(&cert.uuid()))
        .map(|cert| CertWithFormattedString::new(cert, false))
        .collect::<Vec<CertWithFormattedString>>();
```